src/mongocxx/test/search_index_view.cpp

```diff
@@ -250,9 +273,299 @@ TEST_CASE("atlas search indexes prose tests", "") {
+        });
+
+        //   {
+        //     name: 'test-search-index-case7-vector',
+        //     type: 'vectorSearch',
+        //     definition: {
+        //       fields: [
+        //          {
+        //              type: 'vector',
+        //              path: 'plot_embedding',
+        //              numDimensions: 1536,
+        //              similarity: 'euclidean',
+        //          },
+        //       ]
+        //     }
+        //   }
+        const auto name3 = "test-search-index-case7-vector";
+        const auto type3 = "vectorSearch";
+        const auto definition3 =
+            make_document(kvp("fields",
+                              make_array(make_document(kvp("type", "vector"),
+                                                       kvp("path", "plot_embedding"),
+                                                       kvp("numDimensions", 1536),
+                                                       kvp("similarity", "euclidean")))));
+        const auto model3 = search_index_model(name3, definition3.view(), type3);
+
+        REQUIRE(model3.type().value() == "vectorSearch");
+        REQUIRE(siv.create_one(model3) == "test-search-index-case7-vector");
+
+        assert_soon([&siv, &model3](void) -> bool {
+            auto cursor = siv.list();
+            return does_search_index_exist_on_cursor_with_type(
+                cursor, model3, "vectorSearch", false);
+        });
+
+        SUCCEED(
+            "Prose Test Case 7: Driver can successfully handle search index types when creating "
+            "indexes");
+    }
+
+    // Prose test case #8.
+    SECTION("Driver requires explicit type to create a vector search index") {
+        bsoncxx::oid id;
+        auto coll0 = db.create_collection(id.to_string());
+        auto siv = coll0.search_indexes();
+        //   {
+        //     name: 'test-search-index-case8-error',
+        //     definition: {
+        //       fields: [
+        //          {
+        //              type: 'vector',
+        //              path: 'plot_embedding',
+        //              numDimensions: 1536,
+        //              similarity: 'euclidean',
+        //          },
+        //       ]
+        //     }
+        //   }
+        const auto name = "test-search-index-case8-error";
+        const auto definition =
+            make_document(kvp("fields",
+                              make_array(make_document(kvp("type", "vector"),
+                                                       kvp("path", "plot_embedding"),
+                                                       kvp("numDimensions", 1536),
+                                                       kvp("similarity", "euclidean")))));
+        const auto model = search_index_model(name, definition.view());
+
+        REQUIRE_THROWS_WITH(siv.create_one(model),
+                            "Attribute mappings missing.: generic server error");
```

Comment:
```suggestion
        REQUIRE_THROWS_WITH(siv.create_one(model), "Attribute mappings missing");
```
Avoid asserting more than is required by spec.

Reply:
```suggestion
        REQUIRE_THROWS_WITH(siv.create_one(model),
                            Catch::Matchers::Contains("Attribute mappings missing"));
```
Fixed suggestion to match against a substring.